tests/bearlib/aspects/BaseTest.py

```diff
@@ -15,5 +15,7 @@ def test_class(self):
         assert not isinstance(aspectbase, aspectclass)
 
     def test_init_needs_aspectclass(self):
-        with pytest.raises(AttributeError):
+        error_str = 'object has no attribute'
```

Comment:
Just `with self.assertRaises(AttributeError):`

The other two lines are unnecessary, and you don't have a complicated set of Exceptions whose names you want to match via a regular expression, you are just checking for a specific exception raised (so use `assertRaises` instead).